Admin dash deployments

StephenAraka opened the pull request:
### what does this PR do?
get deployments (admin dash)

### screenshots?
- From Design
![Screenshot from 2020-03-02 10-42-17](https://user-images.githubusercontent.com/29985169/75683244-d620f780-5ca7-11ea-940d-0d09a3c6f232.png)

- From implementation
![Screenshot from 2020-03-02 16-54-31](https://user-images.githubusercontent.com/29985169/75683278-e2a55000-5ca7-11ea-8d89-e72a3e0d561f.png)


### How to run this?
- Run the branch

### Question...?
- @Mubangizi  mentioned that the READY column is better implemented as fractions (see example in design). 
- Should this be changed?

### Some more info
### GENERAL STYLING FOR THE TABLE
I've added some general styling for the table... All you need to do is, wrap your table within a **div** with **className="ResourcesTable"** 
Like this... 👇
![Screenshot from 2020-03-02 17-11-29](https://user-images.githubusercontent.com/29985169/75683944-0917bb00-5ca9-11ea-8e54-c2dc3b35058e.png)

Mubangizi commented:
@StephenAraka  the width of the Ready Progress Bar looks too big as opposed to the design.
I think it needs to be reduced

Mubangizi commented:
Okay points to note...
- There is a helper function called AgeUtility, its PR is one for **PODS** 
- Then also the width of the ready column component.
Then will be good?

neelxie commented on package.json:
```diff
@@ -4,6 +4,7 @@
   "private": true,
   "dependencies": {
     "axios": "^0.19.1",
+    "moment": "^2.24.0",
```
@StephenAraka we could have a custom function that replaces what this third party library does, I had one but we refactored it with CK and Allan and agreed on how it ought respond.

neelxie commented on src/redux/actions/getDeployments.js:
```diff
@@ -0,0 +1,20 @@
+const getDeployments = (clusterID) => (dispatch) => {
+  axios.get(`${API_BASE_URL}/clusters/${clusterID}/deployments`)
+    .then((response) => {
+      console.log(response.data.data.deployments);
```
Kindly remove the console logs, we cant have these and the debugger lines in our production code.

neelxie commented on src/redux/actions/getDeployments.js:
```diff
@@ -0,0 +1,20 @@
+  axios.get(`${API_BASE_URL}/clusters/${clusterID}/deployments`)
+    .then((response) => {
+      console.log(response.data.data.deployments);
+      debugger;
```
This line too ought to be removed.

neelxie commented on src/redux/actions/getDeployments.js:
```diff
@@ -0,0 +1,20 @@
+      });
+    })
+    .catch((error) => {
+      console.log(error);
```
We could have an action that handles the error returned instead of a console log given this doesnt cater for users incase an error is returned by the backend end and a user requires an appropriate response.

neelxie commented on src/redux/reducers/getDeployments.js:
```diff
@@ -0,0 +1,20 @@
+      ...state,
+      deployments: action.payload
+    };
+
```
I think we will need for a second case that caters for error messages too,

neelxie commented on src/components/DeploymentsPage/index.js:
```diff
@@ -0,0 +1,101 @@
+                      <tr>
+                        <td>{deployment.metadata.name}</td>
+                        <td>
+                          {Object.prototype.hasOwnProperty.call(deployment.status, 'readyReplicas') ? (
```
The status wrapper you added in @Mubangizi's PR could work as well here.

Mubangizi commented:
Please Use the helper `TellAge` Method for getting days past. its already in staging. **Then You can merge**

StephenAraka commented:
> Please Use the helper `TellAge` Method for getting days past. its already in staging. **Then You can merge**

done this...

StephenAraka commented:
> @StephenAraka Kindly cater for the feedback as stated in the various files, not forgetting the clustername in the side nav.

finished processing the feedback

StephenAraka commented:
### Processed Feedback
1. Reduced width of ProgressBar
2. Changed Deloyments Ready Column from percentage to fractions
3. Removed Console logs and debugger;
4. Added Big Spinner to table
5. Removed moment, now using tellAge helper function

### Testing
Please checkout to branch, and pull changes
 
(Screenshots have jam to upload ☹️)

Mubangizi commented:
> ### Processed Feedback
> 1. Reduced width of ProgressBar
> 2. Changed Deloyments Ready Column from percentage to fractions
> 3. Removed Console logs and debugger;
> 4. Added Big Spinner to table
> 5. Removed moment, now using tellAge helper function
> 
> ### Testing
> Please checkout to branch, and pull changes
> 
> (Screenshots have jam to upload )

Great Work @StephenAraka
